src/context/MainContexts.js

```diff
@@ -0,0 +1,31 @@
+import { createContext } from "react";
+export const API_URL = "https://jsonplaceholder.typicode.com";
+
+export const initialUsersState = {
+  isLoading: true,
+  isError: false,
+  error: "",
+  users: [],
+  user: {},
+};
+export const UsersContext = createContext();
+
+export const initialPostsState = {
+  isLoading: true,
+  modalLoading: true,
+  isError: false,
+  error: "",
+  posts: [],
+  comments: [],
+};
+export const PostsContext = createContext();
+
+export const initialPostCrud = {
```

Comment:
Put this data in a DataHelpers file and export a function named getInitialCridData from there and then utilize the function instead of passing it via context.